test/cctest/test-assembler-riscv64.cc

```diff
@@ -1431,6 +1431,97 @@ TEST(RISCV9) {
   USE(code);
 }
 
+TEST(NAN_BOX) {
+  // Test float NaN-boxing.
+  CcTest::InitializeVM();
+  float neg_inf = -std::numeric_limits<float>::infinity();
+
+  auto fn = [](MacroAssembler& assm) { __ fmv_x_d(a0, fa0); };
+  GenAndRunTest(neg_inf,
+      ((uint64_t)-1 << 32) | bit_cast<int32_t>(neg_inf),
+      fn);
+  Isolate* isolate = CcTest::i_isolate();
+  HandleScope scope(isolate);
+
+  struct T {
+    float a;
+    float b;
+    uint64_t r0;
+    int r1;
+    int r2;
+    int r3;
+    int r4;
+    uint64_t r5;
+    float fr1;
+    float fr2;
+  };
+  T t;
+
+  MacroAssembler assm(isolate, v8::internal::CodeObjectRequired::kYes);
+
+  // Load all structure elements to registers.
+  __ flw(fa0, a0, offsetof(T, a));
+  __ flw(fa1, a0, offsetof(T, b));
+
+  __ fmv_w_x(ft0, zero_reg); // float 0
+  __ fmv_d_x(ft1, zero_reg); // float NaN
+
+  // Test boxing and unbox
+  __ fsw(fa0, a0, offsetof(T, a));
+  __ fsd(fa0, a0, offsetof(T, r0));
+  __ fsw(ft1, a0, offsetof(T, b));
+
+  // Compare
+  __ flt_s(a1, fa0, ft0); 
+  __ sw(a1, a0, offsetof(T, r1));
+  __ flt_s(a1, fa0, ft1); 
+  __ sw(a1, a0, offsetof(T, r2));
+
+  // Compute 
+  __ fadd_s(fa3, fa1, ft0); 
+  __ fsw(fa3, a0, offsetof(T, fr1));
+  __ fadd_s(fa3, fa1, ft1); 
+  __ fsw(fa3, a0, offsetof(T, fr2));
+
+  // Convert float into int
+  __ fcvt_w_s(a1, ft0);
+  __ sw(a1, a0, offsetof(T, r3));
+  __ fcvt_w_s(a1, ft1);
+  __ sw(a1, a0, offsetof(T, r4));
+
+  // Convert int into float 
+  __ fcvt_s_w(fa3, zero_reg);
+  __ fsd(fa3, a0, offsetof(T, r5));
+
+  __ jr(ra);
+
+  CodeDesc desc;
+  assm.GetCode(isolate, &desc);
+  Handle<Code> code = Factory::CodeBuilder(isolate, desc, Code::STUB).Build();
+  auto f = GeneratedCode<F3>::FromCode(*code);
+  t.a = neg_inf;
+  t.b = 1.234f;
+  t.r1 = -1;
+  t.r2 = -1;
+  t.r3 = -1;
+  t.r4 = -1;
+  t.r5 = -1;
+  t.fr1 = -1.0f;
+  t.fr2 = -1.0f;
```

Comment:
This test case is not easy to understand. There is a lot of stuff that is not related to Nan-boxing testing, perhaps things copied from other tests? Could you make it simpler and remove the unrelated tests?

Make sure you also test the nan-boxing of `flw` instructions.